File: data/code_pairs/cpp/417.cpp
#include <iostream>
#include <memory>

class T {
public:
    int dummy = 0;
};

class Main {
public:
    void run() {
        T b; // Stack allocation
        b.dummy = 5;
        T* a = &b; // Pointer
        std::unique_ptr<T> c(new T); // Smart pointer for dynamic allocation

        sub(a);

        if (a == nullptr) {
            std::cout << "Test passed: a is nullptr." << std::endl;
        } else {
            std::cout << "Test failed: a is not nullptr." << std::endl;
        }
    }

private:
    void sub(T*& a) {
        a = nullptr;
        std::cout << "Subroutine SUB was called." << std::endl;
    }
};

int main() {
    Main program;
    program.run();
    return 0;
}

File: data/code_pairs/fortran/417.f90
PROGRAM MAIN
  IMPLICIT NONE
  TYPE :: t
     INTEGER :: dummy = 0
  END TYPE t
  CLASS(t), POINTER :: a
  TYPE(t), TARGET :: b
  CLASS(t), ALLOCATABLE :: c

  b%dummy = 5
  a => b
  CALL SUB(a)

  IF (.NOT. ASSOCIATED(a)) THEN
     PRINT *, "Test passed: a is not associated."
  ELSE
     PRINT *, "Test failed: a is associated."
  END IF

CONTAINS
  SUBROUTINE SUB(a)
    CLASS(t), POINTER :: a
    a => NULL()
    PRINT *, "Subroutine SUB was called."
  END SUBROUTINE SUB
END PROGRAM MAIN